templates/shift-assignment-pyomo/main.py

```diff
@@ -0,0 +1,315 @@
+import argparse
+import datetime
+import json
+import sys
+from typing import Any
+
+import pyomo.environ as pyo
+
+# Duration parameter for the solver.
+SUPPORTED_PROVIDER_DURATIONS = {
+    "cbc": "sec",
+    "glpk": "tmlim",
+}
+
+# Status of the solver after optimizing.
+STATUS = {
+    "suboptimal": "suboptimal",
+    "infeasible": "infeasible",
+    "optimal": "optimal",
+    "unbounded": "unbounded",
+}
+
+
+def main() -> None:
+    """Entry point for the app."""
+
+    parser = argparse.ArgumentParser(
+        description="Solve shift-assignment with Pyomo."
+    )
+    parser.add_argument(
+        "-input",
+        default="",
+        help="Path to input file. Default is stdin.",
+    )
+    parser.add_argument(
+        "-output",
+        default="",
+        help="Path to output file. Default is stdout.",
+    )
+    parser.add_argument(
+        "-duration",
+        default=30,
+        help="Max runtime duration (in seconds). Default is 30.",
+        type=int,
+    )
+    parser.add_argument(
+        "-provider",
+        default="cbc",
+        help="Solver provider. Default is cbc.",
+    )
```

Comment:
This is interesting. We should probably do this for ortools as well, if we do it for pyomo, right?